background.js

```diff
@@ -0,0 +1,471 @@
+
+	const testCaseId = updateRequest.testCaseResult.testCaseId;
+	const updatedIterationActionResult = getUpdatedIterationActionResult(updateRequest);
+	const testRunId = updateRequest.testRunId;
+
+	tabToAdosTestRunUrlMap[tabId] = tab.url.substring(0, tab.url.indexOf('/_testExecution/')) +
+		'/_testManagement/runs?runId=' + testRunId + '&_a=runCharts';
+
+	if (!testParametersMap[testCaseId]) {
+		testParametersMap[testCaseId] = getParameterDefinitionsPerIteration(updateRequest);
+	}
+
+	const testNameWithParameter = updateRequest.testCaseResult.testCaseTitle.trim();
+
+	if (updatedIterationActionResult.outcome === 12) { // test is paused, other call to Teamscale needed
+		queryTeamscale('pause', details.tabId, testNameWithParameter,
+			testOutcomeToTeamscaleTestExecutionResult(updatedIterationActionResult.outcome));
+	} else {
+		queryTeamscale('update', details.tabId, testNameWithParameter,
+			testOutcomeToTeamscaleTestExecutionResult(updatedIterationActionResult.outcome));
+	}
+}
+
+/**
+ * Object returned carries property "outcome" that describes the testing result as integer which can be interpreted by
+ * {@link testOutcomeToTeamscaleTestExecutionResult}.
+ */
+function getUpdatedIterationActionResult(updateRequest) {
+	let result;
+
+	for (let i = 0; i < updateRequest.actionResults.length; i++) {
+		const actionResult = updateRequest.actionResults[i];
+
+		if (isSubstepInfo(actionResult)) {
+			continue;
+		}
+
+		if (!result || getCompletedDateOfActionResult(result) < getCompletedDateOfActionResult(actionResult)) {
+			result = actionResult;
+		}
+	}
+
+	return result;
+}
+
+function getParameterDefinitionsPerIteration(updateRequest) {
+	let parametersPerIteration = {};
+
+	for (let i = 0; i < updateRequest.parameters.length; i++) {
+		const parameter = updateRequest.parameters[i];
+
+		if (isSubstepInfo(parameter)) {
+			continue;
+		}
+
+		if (!parametersPerIteration[parameter.iterationId]) {
+			parametersPerIteration[parameter.iterationId] = new Set();
+		}
+
+		const parameterNameAndValue = parameter.parameterName + '=' + parameter.expected;
+		parametersPerIteration[parameter.iterationId].add(parameterNameAndValue);
+	}
+
+	return parametersPerIteration;
+}
+
+function queryTeamscale(action, tabId, extendedName, status, sapTestKey) {
+	if (action !== 'reset' && action !== 'log' && (!tabId || !tabToTestCaseMap[tabId])) {
+		throw 'Could not obtain testId from tabId "' + tabId + '".';
+	}
+
+	if (!configOptions[tsServerOptionName] || !configOptions[tsProjectOptionName] || !configOptions[sapUserOptionName]) {
+		throw 'Not all extension configuration entries are set. (' +
+		tsServerOptionName + '=' + configOptions[tsServerOptionName] + ', ' +
+		tsProjectOptionName + '=' + configOptions[tsProjectOptionName] + ', ' +
+		sapUserOptionName + '=' + configOptions[sapUserOptionName] + ')';
+	}
+
+	const testId = tabToTestCaseMap[tabId];
+
+	if (action === 'update' && (!extendedName || !status)) {
```

Comment:
Maybe extracting this `update` and `reset` values as a **global** enum? I don't know how we can do this in JS but I think it would be nice if we have one. Therefore we don't need to think too much about the places where these actions are being used and whether they are correctly defined

Reply:
extracted all actions to a constant in `constants.js`